framework/db/ExpressionBuilder.php

```
@@ -25,7 +25,27 @@ class ExpressionBuilder implements ExpressionBuilderInterface
+            ARRAY_FILTER_USE_BOTH
+        );
+        foreach (array_keys($duplicateKeys) as $duplicateKey) {
+            $existingKey = array_search($newParams[$duplicateKey], $params, true);
+            if ($existingKey !== false) {
+                // we already have this value in our params, so just re-use it to avoid wasted space
+                $newKey = $existingKey;
+            } else {
+                // use an arbitrary key to avoid clashing
+                $newKey = $duplicateKey . count($params);
+                $newParams[$newKey] = $newParams[$duplicateKey];
+            }
```

PHPMD (2.15.0)

[error] The method build uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them. (undefined)

(ElseExpression)